feat: 프로필 페이지

yeni023 opened the pull request:
## 🔗 반영 브랜치

ex. (#119) yeeun/Profile -> dev

## 📝 작업 내용
ProfilePage 사용자 정보 UI 구현

### 📸 스크린샷 (선택)
<img width="792" alt="스크린샷 2025-05-08 오후 10 24 52" src="https://github.com/user-attachments/assets/1d454fda-7d40-4b8b-a65e-675a3ae8e5b8" />

## 💬 리뷰 요구사항(선택)
- 기존에 작업한 파일 코드들 말고 ui 구현부터 다시 해봤습니다 파일 분류도 이렇게 나누는 게 맞는 건지 살짝 헷갈려서 한 번 더 확인해주시면 좋을 것 같아요
- Slack 에서 추천해주신 DungGeunMo 폰트를 전역으로 사용했는데 저 위치에 폰트를 추가하는 게 맞는지, 사용하는 방법도 맞게 했는지 한 번 확인해주실 수 있을까요?
- 지금 작업한 부분은 UI 배치만 일단 작업했고 버튼 클릭했을 때 뜨는 모달과 값 받아오는 api 도 빠르게 개발 진행하겠습니다.

yeni023 committed:
feat: 프로필 페이지

hakilee commented on src/pages/ProfilePage/components/profile-card/index.tsx:
```diff
@@ -0,0 +1,35 @@
+import LogoutButton from '@/pages/ProfilePage/components/logout-button';
```
```suggestion
import LogoutButton from './components/logout-button';
```

hakilee commented on src/pages/ProfilePage/components/profile-card/index.tsx:
```diff
@@ -0,0 +1,35 @@
+import LogoutButton from '@/pages/ProfilePage/components/logout-button';
+import ProfileImage from '@/pages/ProfilePage/components/profile-image';
```
```suggestion
import ProfileImage from './components/profile-image';
```

하위 컴포넌트이기 때문에 상대경로로 적어주는 것이 조금 더 이해가 쉬운 것 같아요.

hakilee reviewed (commented):
고생하셨습니다! 몇 가지 사소한 부분에 의견을 말씀드릴게요.

1. 현재 스타일을 가져와서 사용할 때,

```typescript
import * as S from './styles.css';
```
로 사용하고 계신데요, 취향 차이라고 생각되지만, `S`라는 의미가 와닿지 않는 것 같아요. 확실하게 `styles` 라는 값으로 사용하는 것은 어떨까요? [이 글](https://velog.io/@mowinckel/%EB%88%84%EA%B0%80-%EC%9D%B4%EB%A6%84%EC%9D%84-%ED%95%A8%EB%B6%80%EB%A1%9C-%EC%A7%93%EB%8A%94%EA%B0%80)을 한 번 참고해 주세요.

2. 폰트 가져오는 로직은 제가 생각한 것과 같습니다. 다만, 현재 `ttf` 파일을 사용하고 계신데, 웹 최적화 폰트인 `woff2` 확장명을 가진 파일로 변경해 주세요.

hakilee commented on src/pages/ProfilePage/components/profile-card/index.tsx:
```diff
@@ -0,0 +1,35 @@
+
+  return (
+    <S.CardWrapper>
+      <S.CloseButton>×</S.CloseButton>
```
이 부분도 이미 추가되어 있는 이미지로 변경해주세요

yeni023 committed:
chore: 리뷰 반영에 따른 import 및 UI 개선

- 스타일 import 시 S → styles 명확화
- 폰트 파일 확장자 ttf → woff2로 교체
- close 버튼의 텍스트(×)를 SVG 이미지로 교체
- 하위 컴포넌트 import를 절대경로 → 상대경로로 변경

hakilee committed:
refactor: 라우트 구조 변경: 비로그인/로그인 라우트 분리

로그인 여부에 따라 접근 가능한 페이지를 PrivateRoute로 묶어 관리합니다.
`/history`, `/profile` 페이지를 로그인된 사용자만 접근 가능하도록 변경했습니다.
비로그인 사용자는 해당 페이지에 접근할 수 없습니다.

hakilee committed:
style: 프로필 페이지 스타일 조정 및 불필요한 속성 제거

- 프로필 카드 스타일에서 불필요한 `box-sizing: border-box;` 속성 제거
- 프로필 카드 텍스트 스타일에서 폰트 관련 속성 제거 (상위 스타일에서 관리하도록 변경)
- 프로필 페이지 스타일에서 불필요한 `box-sizing: border-box;`와 `min-height: 100vh;` 속성 제거
